### DISCUSSION TITLE
Question: When to use which lense?

### DISCUSSION THREAD

**Discussion by alexHeu** [MLWave/kepler-mapper#259] [Q&A]
Hi,

I just wanted to ask whether there are some general guidelines/resources/examples/best practices on when to use which lense(s)?

My specific use-cases which I have in mind are customer segmentations and exploratory analysis to distinguish customer groups that have a high/low customer lifetime value.
Potential inputs for the segmentation are LRFM features (L - lenght of the customer relationship, R - time between now and last purchase, F - purchase frequency, M - average transaction amount).

Right now I am not yet quite sure which lenses or combination of lenses might make sense. Therefore I wanted to ask whether some of you already tackled similar use-cases and has some experiences to share or some general guidelines that might help me before I start experimenting.

Thanks in advance and best regards
Alex

**Comment by MLWave** [comment]
Use one of these in combination with:
- a density estimator (Gauss density, ...)
- outlier detector (NNeighbour distance, isolation forest, ...)
- dimensionality reduction component (SVD, NMF, UMAP, ...)

Or combinations:

[Purchase frequency, Average transaction amount]

If you have a predictive model with high performance on Customer Lifetime Value, you can incorporate the output of this model as a lens too.
